flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/strategy/RequestInfo.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.sink.writer.strategy;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/** Dataclass to encapsulate information about starting / completing requests. */
+@PublicEvolving
+public class RequestInfo {
+    public final int failedMessages;
+    public int batchSize;
+    public long requestStartTime;
+
+    private RequestInfo(int failedMessages, int batchSize, long requestStartTime) {
+        this.failedMessages = failedMessages;
+        this.batchSize = batchSize;
+        this.requestStartTime = requestStartTime;
+    }
+
+    public void setBatchSize(int batchSize) {
+        this.batchSize = batchSize;
+    }
+
+    public void setRequestStartTime(long requestStartTime) {
+        this.requestStartTime = requestStartTime;
+    }
+
+    @PublicEvolving
+    public static RequestInfoBuilder builder() {
+        return new RequestInfoBuilder();
+    }
+
+    /** Builder for {@link RequestInfo} dataclass. */
+    public static class RequestInfoBuilder {
+        private int failedMessages;
+        private int batchSize;
+        private long requestStartTime;
+
```

Comment:
The fact these fields might be `0` under certain conditions which means "not set" reinforces the previous concerns I have mentioned. If we genuinely want to reuse this object and leverage setters we would need to distinguish between `failedMessages` as 0 or unset. Could be achieved with `-1`, but prefer refactor